app/api/controllers/RequestHandler.scala

```diff
@@ -109,19 +109,28 @@ object RequestHandler {
           message = s"[${ctx.endpointLogContext.controllerName}][${ctx.endpointLogContext.endpointName}] " +
             s"with correlationId : ${ctx.correlationId}")
 
-        val result =
-          for {
-            parsedRequest   <- EitherT.fromEither[Future](parser.parseRequest(rawData))
-            serviceResponse <- EitherT(service(parsedRequest))
-          } yield doWithContext(ctx.withCorrelationId(serviceResponse.correlationId)) { implicit ctx: RequestContext =>
-            handleSuccess(rawData, parsedRequest, serviceResponse)
-          }
-
-        result.leftMap { errorWrapper =>
-          doWithContext(ctx.withCorrelationId(errorWrapper.correlationId)) { implicit ctx: RequestContext =>
+        val maybeGovTestScenario = ctx.hc.otherHeaders.find(header => header._1 == "Gov-Test-Scenario").map(headers => headers._2).getOrElse("")
+        if (maybeGovTestScenario == "REQUEST_CANNOT_BE_FULFILLED") {
+          val errorWrapper = ErrorWrapper(ctx.correlationId, RuleRequestCannotBeFulfilled)
+          Future.successful(doWithContext(ctx.withCorrelationId(errorWrapper.correlationId)) { implicit ctx: RequestContext =>
             handleFailure(errorWrapper)
-          }
-        }.merge
+          })
+        } else {
+          val result =
+            for {
+              parsedRequest   <- EitherT.fromEither[Future](parser.parseRequest(rawData))
+              serviceResponse <- EitherT(service(parsedRequest))
+            } yield doWithContext(ctx.withCorrelationId(serviceResponse.correlationId)) { implicit ctx: RequestContext =>
+              handleSuccess(rawData, parsedRequest, serviceResponse)
+            }
```

Comment:
Can replace lines 112 down to 125 with this:
```
        val clientRequestedError =
          ctx.hc.otherHeaders.contains("Gov-Test-Scenario" -> "REQUEST_CANNOT_BE_FULFILLED")

        val result =
          if (clientRequestedError)
            EitherT(
              Future.successful(Left(ErrorWrapper(ctx.correlationId, RuleRequestCannotBeFulfilled)))
            )
          else
            for {
              parsedRequest   <- EitherT.fromEither[Future](parser.parseRequest(rawData))
              serviceResponse <- EitherT(service(parsedRequest))
            } yield doWithContext(ctx.withCorrelationId(serviceResponse.correlationId)) { implicit ctx: RequestContext =>
              handleSuccess(rawData, parsedRequest, serviceResponse)
            }
```
& keep the result.leftMap.

Reply:
result.leftmap as it is returns an ANY which is an invalid return for handlerequest